django_mongodb/compiler.py

```diff
@@ -139,44 +310,22 @@ def check_query(self):
             if any(key.startswith("_prefetch_related_") for key in self.query.extra):
                 raise NotSupportedError("QuerySet.prefetch_related() is not supported on MongoDB.")
             raise NotSupportedError("QuerySet.extra() is not supported on MongoDB.")
-        if any(
-            isinstance(a, Aggregate) and not isinstance(a, Count)
-            for a in self.query.annotations.values()
-        ):
-            raise NotSupportedError("QuerySet.aggregate() isn't supported on MongoDB.")
-
-    def get_count(self, check_exists=False):
-        """
-        Count objects matching the current filters / constraints.
-
-        If `check_exists` is True, only check if any object matches.
-        """
-        kwargs = {}
-        # If this query is sliced, the limits will be set on the subquery.
-        inner_query = getattr(self.query, "inner_query", None)
-        low_mark = inner_query.low_mark if inner_query else 0
-        high_mark = inner_query.high_mark if inner_query else None
-        if low_mark > 0:
-            kwargs["skip"] = low_mark
-        if check_exists:
-            kwargs["limit"] = 1
-        elif high_mark is not None:
-            kwargs["limit"] = high_mark - low_mark
-        try:
-            return self.build_query().count(**kwargs)
-        except EmptyResultSet:
-            return 0
 
     def build_query(self, columns=None):
         """Check if the query is supported and prepare a MongoQuery."""
         self.check_query()
-        query = self.query_class(self, columns)
+        query = self.query_class(self)
+        query.aggregation_pipeline = self.get_aggregation_pipeline()
         query.lookup_pipeline = self.get_lookup_pipeline()
+        query.order_by(self._get_ordering())
+        query.project_fields = self.get_project_fields(columns, ordering=query.ordering)
         try:
-            query.mongo_query = {"$expr": self.query.where.as_mql(self, self.connection)}
+            where = getattr(self, "where", self.query.where)
```

Comment:
This fallback is a little opaque. It looks to me `self.query.where` is used for Update and Delete compilers. I would either add a comment about this or add a `get_where()` hook to the compilers that returns the appropriate value.

Reply:
Yes, the where from query is very confusing, sometimes it is the same where that camper has, other times it has a where with the `having` expressions. The compiler (the main one) uses self.where, that has the expressions separated. but add a get_where in the compiler could be good enough.